File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        ArrayList<ArrayList<Integer>> arr = new ArrayList<>();

        for(int i=0;i<matrix.length;i++)
        {
            for(int j=0;j<matrix[0].length;j++)
            {
                if(matrix[i][j] == 0)
                {
                    arr.add(new ArrayList<>(Arrays.asList(i,j)));
                }
            }
        }

        for(ArrayList<Integer> pos : arr)
        {
            int ipos = pos.get(0);
            int jpos = pos.get(1);

            for(int j=0;j<matrix[0].length;j++)
            {
                matrix[ipos][j] = 0;
            }

             for(int i=0;i<matrix.length;i++)
            {
                matrix[i][jpos] = 0;
            }
        }

    }
}

File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {

        vector<pair<int, int>> arr;

        for (int i = 0; i < matrix.size(); i++) {
            for (int j = 0; j < matrix[0].size(); j++) {
                if (matrix[i][j] == 0) {
                    arr.push_back({i, j});
                }
            }
        }

        for (auto pos : arr) {
            int ipos = pos.first;
            int jpos = pos.second;

            for (int i = 0; i < matrix.size(); i++)
            {
                matrix[i][jpos] = 0;
            }

            for(int j=0; j< matrix[0].size();j++)
            {
                matrix[ipos][j] = 0;
            }
        }
    }
};